prover.go

```diff
@@ -25,7 +25,7 @@ type Prover struct {
 	l2Config           *types.L2ConfigInfo
 	l1BlockHashOracle  common.Address
 	srcChainID         *big.Int
-	configProof        *types.UpdateL2ConfigArgs
+	configProof        func(blockNum *big.Int) (*types.UpdateL2ConfigArgs, error)
```

Comment:
Nitpick but rather than inlining a func here what if we replace `configProof` with `registryProver`, and then on line 162 call `p.registryProver.GenerateUpdateL2ConfigArgs(ctx, p.srcChainID, blockNum)`?